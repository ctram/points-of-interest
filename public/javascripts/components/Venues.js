import React, { useState } from "react";

import Venue from "./Venue";

import "../../stylesheets/Venues.css";

export default function Venues(props) {
  const { venues } = props;

  let domVenues = null;

  if (venues.length > 0) {
    domVenues = (
      <ul className="venues__list">
        {venues.map((venue, idx) => {
          return <Venue venue={venue} key={venue.id || idx} />;
        })}
      </ul>
    );
  }

  return <div className="venues">{domVenues}</div>;
}
