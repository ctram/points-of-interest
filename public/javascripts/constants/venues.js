const SET_VENUES = "SET_VENUES";

export { SET_VENUES };
