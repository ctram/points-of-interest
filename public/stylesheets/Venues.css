.venues {
  padding-top: 50px;
}

.venues .venues__list {
  display: flex;
  flex-direction: column;
  align-items: center;
}
