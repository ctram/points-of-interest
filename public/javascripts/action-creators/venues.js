import { SET_VENUES } from "../constants/venues";

import FourSquareHelper from "../lib/four-square-helper";

const fourSquareClient = new FourSquareHelper();

export function fetchVenues({ coordinates, near, query, limit }) {
  return dispatch => {
    return fourSquareClient
      .nearByVenues({ coordinates, near, query, limit })
      .then(venues => {
        return dispatch({ type: SET_VENUES, venues });
      })
      .catch(e => {
        console.error(e);
      });
  };
}
