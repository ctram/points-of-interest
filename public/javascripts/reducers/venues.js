import { SET_VENUES } from "../constants/venues";

const initialState = [];

export default function venues(state = initialState, action) {
  const { type, venues } = action;

  switch (type) {
    case SET_VENUES:
      return venues;
    default:
      return state;
  }
}
