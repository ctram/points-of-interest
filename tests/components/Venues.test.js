import React from "react";
import Venues from "../../public/javascripts/components/Venues";
import renderer from "react-test-renderer";

import venues from "../mock-data/venues";

test("Venues component displays a list of venues", () => {
  const component = renderer.create(<Venues venues={venues} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
