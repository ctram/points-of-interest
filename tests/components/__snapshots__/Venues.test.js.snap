// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venues component displays a list of venues 1`] = `
<div
  className="venues"
>
  <ul
    className="venues__list"
  >
    <div
      className="venue card"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Hot Shot
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted venue__tags"
        >
          <span
            className="badge badge-secondary venue__tag"
          >
            Cafe
          </span>
          <span
            className="badge badge-secondary venue__tag"
          >
            Bar
          </span>
        </h6>
        <div
          className="address card-text"
        >
          <div>
            123 Somewhere
          </div>
          <div>
            Gotham, NY 12345
          </div>
          <div />
        </div>
      </div>
    </div>
    <div
      className="venue card"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Drunken Seagull
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted venue__tags"
        >
          <span
            className="badge badge-secondary venue__tag"
          >
            Bar
          </span>
        </h6>
        <div
          className="address card-text"
        >
          <div>
            348 Fall
          </div>
          <div>
            Gotham, NY 12345
          </div>
          <div />
        </div>
      </div>
    </div>
    <div
      className="venue card"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Wandering Bull
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted venue__tags"
        >
          <span
            className="badge badge-secondary venue__tag"
          >
            Restaurant
          </span>
          <span
            className="badge badge-secondary venue__tag"
          >
            Bar
          </span>
        </h6>
        <div
          className="address card-text"
        >
          <div>
            A st.
          </div>
          <div>
            Gotham, NY 12345
          </div>
          <div />
        </div>
      </div>
    </div>
  </ul>
</div>
`;
